ffi/dune

```diff
@@ -0,0 +1,21 @@
+(library
+ (name yaml_ffi)
+ (public_name yaml.ffi)
+ (libraries sexplib)
+ (foreign_archives ../vendor/yaml_c)
+ (ctypes
+  (external_library_name yaml)
+  (build_flags_resolver
+   (vendored
+     ; hack: repeated include dirs from root and relative since this gets
+     ;  called from multiple places
+    (c_flags (-I../vendor/headers -Ivendor/headers))))
+  (headers (preamble "#if defined(__MINGW32__) || defined(__MINGW64__)
```

Comment:
This preamble might be better off in a separate file, and referenced from here. I can see it getting awkward fast to embed C headers in dune files.

Reply:
Oh, the dune stanza already supports that.  I just pushed a version that does the include approach instead.

(I think it's still nice to provide an option to include a preamble in the dune feature as well)